docs/zed/README.md

```diff
@@ -498,598 +860,38 @@ using the "lake" format, e.g.,
-amount of index state for each data object.  Moreover, the design relies on
-indexes only to enhance performance, not to implement the lake semantics.
-Thus, indexes need not exist to operate and can be incrementally added or
-deleted without large indexing jobs needing to rebuild a monolithic index
-after each configuration change.
-
-To optimize pool scans, the lake design relies on the well-known pruning
-concept to skip any data object that the planner determines can be skipped
-based on one or more indexes of that object.  For example, if an object
-has been indexed for field "foo" and the query
 ```
-foo == "bar" | ...
+zed use <commitish>
 ```
-is run, then the scan will consult the "foo" index and skip the data object
-if the value "bar" is not in that index.
+The `use` command sets set the working branch to the indicated commitish.
 
-This approach works well for "needle in the haystack"-style searches.  When
-a search hits every object, this style of indexing would not eliminate any
-objects and thus does not help.
-
-While an individual index lookup involves latency to cloud storage to lookup
-a key in each index, each lookup is cheap and involves a small amount of data
-and the lookups can all be run in parallel, even from a single node, so
-the scan schedule can be quickly computed in a small number of round-trips
-(that navigate very wide B-trees) to cloud object storage.
-
-### Index Rules
-
-An index of an object is created by applying an _index rule_ to a data object
-and recording the binding to the pool's commit journal.  Once the index is
-available, the query planner can use it to optimize Zed lake scans.
-
-Rules come in three flavors:
-* field rule - index all values of a named field
-* type rule - index all values of all fields of a given type
-* aggregation rule - index any results computed by any Zed script run
-over the data object and keyed by one or more named fields, typically used
-to compute partial aggregations.
-
-Rules are organized into groups by name and defined at the lake level
-so that any named group of rules can be applied to data objects from
-any pool.  The group name provides no meaning beyond a reference to
-a set of index rules at any given time.
-
-Rules are created with `zed index create`,
-deleted with `zed index drop`, and applied with
-`zed index apply`.
-
-#### Field Rule
-
-A field rule indicates that all values of a field be indexed.
 For example,
 ```
-zed index create IndexGroupEx field foo
-```
-adds a field rule for field `foo` to the index group named `IndexGroupEx`.
-This rule can then be applied to a data object having a given `<tag>`
-in a pool, e.g.,
-```
-zed index apply -use logs@main IndexGroupEx <tag>
+zed use logs
 ```
-The index is created and a transaction put (somewhere).  Once this transaction
-has been committed to the pool's journal, the index is available for use
-by the query planner.
+provides a "pool-only" commitish that sets the working branch to `logs@main`.
```

Comment:
This is not currently true. For example:

```
$ zed ls
foobar 26ksg4Pf28hSmzvY5j8BwAyncfi key ts order desc
logs 26l5FL2XXmy2U3yQqMEWre6Ovt9 key ts order desc

$ zed use logs
"logs": branch not found
```

At the moment the `-p` option is used to specify just a pool name.

Reply:
#3744 implements what's described here.